mlflow/pipelines/steps/ingest/datasets.py

```diff
@@ -121,7 +121,7 @@ def _get_required_config(cls, dataset_config: Dict[str, Any], key: str) -> Any:
         except KeyError:
             raise MlflowException(
                 f"The `{key}` configuration key must be specified for dataset with"
-                f" format '{dataset_config.get('format')}'"
+                f" using '{dataset_config.get('using')}' format"
```

Comment:
Should we also throw if unrecognized configs are supplied (e.g. user specifies `using: parquet` and also includes `sql: SELECT * FROM ...`)

@mengxr thoughts?

Reply:
Spoke offline that we could add that but maybe we can add this validation outside of the scope of this PR. :) Thank you for the suggestion though. It is really good. We should add that